LibraBFT/ImplShared/Util/Crypto.agda

```diff
@@ -30,8 +30,11 @@ module LibraBFT.ImplShared.Util.Crypto where
   open WithCryptoHash sha256 sha256-cr
 
   -- IMPL-DIFF
-  -- 1. cannot do "qcParts" because bs-concat does returns BitString
-  -- 2. no need to deal with a 'Proposal' for stable test hashes
+  -- 1. Cannot do equivalent of Haskell "qcParts"
```

Comment:
These changes did not quite satisfy be because the first comment still refers to things that can't be understood without seeing the Haskell code (`qcParts`) and because capitalising the first word of the second comment did not sufficiently enhance my understanding of it 😛 .

I looked at the Haskell code and tried to write comments that would satisfy me, but found deeper issues in that our Agda implementation does not accurately reflect the Haskell code, including that we don't yet include `HashTag`s when hashing.

So I have pushed a few commits that aim to explain the issues, make progress towards addressing them, and leave appropriate `TODO`s.

Reply:
@cwjnkins please be aware of the changes I have suggested.  I suspect the lack of `HashTag`s raises some larger issues.  In particular, off the top of my head, it seems that we should also include `Block` as a `Part` when instantiating the system model, and when we do so, we will not be able to prove many of our properties without addressing the issues I've raised in the comments I've pushed, exactly because we would not be able to preclude the "masquerading" possibility mentioned in those comments.  Actually, I've pushed another commit making this point as well.

Reply:
> @cwjnkins please be aware of the changes I have suggested. I suspect the lack of HashTags raises some larger issues.

@mark-moir Understood, thanks for pointing this out!